exchanges/kraken/kraken.go

```diff
@@ -1054,3 +1067,55 @@ func (k *Kraken) GetWebsocketToken() (string, error) {
 	}
 	return response.Result.Token, nil
 }
+
+func (a *assetTranslatorStore) LookupAltname(target string) string {
+	if !a.Seeded() {
+		return ""
+	}
+
+	a.l.Lock()
+	defer a.l.Unlock()
+	alt, ok := a.Assets[target]
+	if !ok {
+		return ""
+	}
+	return alt
+}
+
+func (a *assetTranslatorStore) LookupCurrency(target string) string {
+	if !a.Seeded() {
+		return ""
+	}
+
+	a.l.Lock()
+	defer a.l.Unlock()
+	for k, v := range a.Assets {
+		if v == target {
+			return k
+		}
+	}
+	return ""
+}
+
+func (a *assetTranslatorStore) Seed(orig, alt string) {
+	a.l.Lock()
+	defer a.l.Unlock()
+
+	if a.Assets == nil {
+		a.Assets = make(map[string]string)
+	}
+
+	_, ok := a.Assets[orig]
+	if ok {
+		return
+	}
+
+	a.Assets[orig] = alt
+}
+
+func (a *assetTranslatorStore) Seeded() bool {
+	a.l.Lock()
+	isSeeded := len(a.Assets) > 0
+	a.l.Unlock()
```

Comment:
We can drop the locks and nest under the locks in `LookupCurrency` & `LookupAltname` so we reduce mutex calls.

Reply:
Great pickup, this is good if the map is not populated but not if it is (which it what it will be 99% of the time)